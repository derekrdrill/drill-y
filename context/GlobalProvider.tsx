import * as React from 'react';

import GlobalContext from './GlobalContext';
import GlobalReducer from './GlobalReducer';
import GlobalState from './GlobalState';

import { getAlbums } from './GlobalActions';
interface GlobalProviderProps {
  children: React.ReactNode;
}

const GlobalProvider = ({ children, ...props }: GlobalProviderProps) => {
  const [state, dispatch] = React.useReducer(GlobalReducer, GlobalState);
  return <GlobalContext.Provider value={{ state, dispatch }}>{children}</GlobalContext.Provider>;
};

export default GlobalProvider;
