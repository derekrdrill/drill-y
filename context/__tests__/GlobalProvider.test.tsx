import * as React from 'react';
import * as renderer from 'react-test-renderer';

import GlobalProvider from '../GlobalProvider';

describe('GlobalProvider unit tests', () => {
  it('renders as expected', () => {
    const result = renderer.create(<GlobalProvider>{<h1>TEST</h1>}</GlobalProvider>).toJSON();
    expect(result).toMatchSnapshot();
  });
});
