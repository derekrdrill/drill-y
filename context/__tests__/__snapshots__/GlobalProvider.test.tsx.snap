// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalProvider unit tests renders as expected 1`] = `
<h1>
  TEST
</h1>
`;
